media-libs/opencv/opencv-4.12.0.recipe

```diff
@@ -0,0 +1,476 @@
+	fi
+
+	pythonVersion=$(python3 --version 2>&1 | sed 's/Python //' | head -c4)
+
+	cmake .. \
+		-DCMAKE_BUILD_TYPE=RelwithDebInfo \
+		$cmakeDirArgs \
+		-DBUILD_TESTS=OFF \
+		-DBUILD_PERF_TESTS=OFF \
+		-DCMAKE_INSTALL_BINDIR=$commandBinDir \
+		-DCMAKE_INSTALL_LIBEXECDIR=$libDir \
+		-DWITH_GDAL=ON \
+		-DWITH_IPP=OFF \
+		-DWITH_OPENGL=OFF \
+		-DWITH_QT=ON \
+		-DWITH_EIGEN=ON \
+		-DWITH_OPENEXR=ON \
+		-DWITH_FFMPEG=ON \
+		-DCMAKE_SKIP_RPATH=ON \
+		-DENABLE_PIC=OFF \
+		-DOPENCV_GENERATE_PKGCONFIG=YES \
+		-DCMAKE_INSTALL_PREFIX:PATH="$prefix" \
+		-DOPENCV_OTHER_INSTALL_PATH:PATH="$dataDir" \
+		-DOpenBLAS_INCLUDE_DIR="$portPackageLinksDir/openblas${secondaryArchSuffix}_devel/develop/headers/$ARCH_SUBDIR" \
+		-DOpenBLAS_LIB="$portPackageLinksDir/openblas${secondaryArchSuffix}_devel/develop/lib/$ARCH_SUBDIR" \
+		-DOPENCV_PYTHON3_INSTALL_PATH="$prefix/lib/python$pythonVersion/vendor-packages" \
+		-DPYTHON3_NUMPY_INCLUDE_DIRS=`finddir B_SYSTEM_LIB_DIRECTORY`/python$pythonVersion/vendor-packages/numpy/_core/include \
+		-DOPENCV_EXTRA_MODULES_PATH="$sourceDir2/opencv_contrib-$portVersion/modules" \
+		-DEigen3_DIR=`finddir B_SYSTEM_DATA_DIRECTORY`/eigen3/cmake
+
+	make $jobArgs
+}
+
+INSTALL()
+{
+	cd build_haiku
+
+	make install
+
+	prepareInstalledDevelLibs libopencv_alphamat \
+		libopencv_aruco \
+		libopencv_bgsegm \
+		libopencv_bioinspired \
+		libopencv_calib3d \
+		libopencv_ccalib \
+		libopencv_core \
+		libopencv_cvv \
+		libopencv_datasets \
+		libopencv_dnn_objdetect \
+		libopencv_dnn_superres \
+		libopencv_dnn \
+		libopencv_dpm \
+		libopencv_face \
+		libopencv_features2d \
+		libopencv_flann \
+		libopencv_freetype \
+		libopencv_fuzzy \
+		libopencv_gapi \
+		libopencv_hdf \
+		libopencv_hfs \
+		libopencv_highgui \
+		libopencv_img_hash \
+		libopencv_imgcodecs \
+		libopencv_imgproc \
+		libopencv_intensity_transform \
+		libopencv_line_descriptor \
+		libopencv_mcc \
+		libopencv_ml \
+		libopencv_objdetect \
+		libopencv_optflow \
+		libopencv_phase_unwrapping \
+		libopencv_photo \
+		libopencv_plot \
+		libopencv_quality \
+		libopencv_rapid \
+		libopencv_reg \
+		libopencv_rgbd \
+		libopencv_saliency \
+		libopencv_shape \
+		libopencv_stereo \
+		libopencv_stitching \
+		libopencv_structured_light \
+		libopencv_superres \
+		libopencv_surface_matching \
+		libopencv_text \
+		libopencv_tracking \
+		libopencv_video \
+		libopencv_videoio \
+		libopencv_videostab \
+		libopencv_xfeatures2d \
+		libopencv_ximgproc \
+		libopencv_xobjdetect \
+		libopencv_xphoto \
+		libopencv_wechat_qrcode
+
+	# fixCmake
+	sed -i 's,\/include,/'${relativeIncludeDir}',g' \
+		$libDir/cmake/opencv4/OpenCVConfig.cmake
+
+	fixPkgconfig
+
+	# fixPkgconfig
+	if [ -n "$secondaryArchSuffix" ]; then
+		sed -i 's,\/headers/x86,\/headers/x86/opencv4,g' \
+			$prefix/$relativeDevelopLibDir/pkgconfig/opencv4.pc
+	else
+		sed -i 's,\/headers,\/headers/opencv4,g' \
+			$prefix/$relativeDevelopLibDir/pkgconfig/opencv4.pc
+	fi
```

Comment:
This could also be simplified with `$secondaryArchSubDir` (and with that also support other potential secondary architectures than x86).